Makefile

```diff
@@ -35,3 +35,15 @@ $(call add-bindata,etcd,./bindata/etcd/...,bindata,etcd_assets,pkg/operator/etcd
 # See vendor/github.com/openshift/build-machinery-go/scripts/run-telepresence.sh for usage and configuration details
 export TP_DEPLOYMENT_YAML ?=./manifests/0000_12_etcd-operator_06_deployment.yaml
 export TP_CMD_PATH ?=./cmd/cluster-etcd-operator
+
+# This was copied from https://github.com/openshift/machine-config-operator
+# Run tests
+test: test-unit test-e2e
+
+# Unit tests only (no active cluster required)
+test-unit:
+	CGO_ENABLED=0 go test -tags=$(GOTAGS) -count=1 -v ./cmd/... ./pkg/... 
```

Comment:
nit:

```console
$ git --no-pager show --oneline --check origin/pr/394
f611f340 (origin/pr/394) Add e2e test framework and add a test for EtcdQuorumGuard
Makefile:45: trailing whitespace.
+	CGO_ENABLED=0 go test -tags=$(GOTAGS) -count=1 -v ./cmd/... ./pkg/... 
```

Reply:
@retroflexer I brought this up but I am not sure why we must redefine test-unit. KAS has both unit and e2e tests.

Reply:
```
vendor/github.com/openshift/build-machinery-go/make/golang.mk
```

Reply:
@hexfusion We need to overwrite the default rules. The default test-unit is attempting to run e2e tests as well, and was failing. By specifying only ./cmd and ./pkg directories, we are excluding it from running the tests in ./e2e directory.

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/394/pull-ci-openshift-cluster-etcd-operator-master-unit/1285873347153891328

Reply:
I deleted these overriding rules, and followed the examples from  the Makefile of https://github.com/openshift/cluster-kube-apiserver-operator. Now we have to wait for the all the tests to pass again :-).